### ISSUE TITLE
Mono: Issue building godot on windows (KeyError: mono_prefix)

### ISSUE THREAD

**Issue by exts** [godotengine/godot#29455]
**Godot version:**
Master

**OS/device including version:**
Windows 10

**Mono**
5.18.x

**MSBuild Tools**
x64 2017

**Issue description:**
```
[Initial build] mono_build_solution(["bin\GodotSharp\Tools\GodotSharpTools.dll"], ["modules\mono\editor\GodotSharpTools\GodotSharpTools.sln"])
scons: *** [bin\GodotSharp\Tools\GodotSharpTools.dll] KeyError : 'mono_prefix'
Traceback (most recent call last):
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Action.py", line 1197, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "D:\godot-master\build\godot\modules\mono\build_scripts\godotsharptools_build.py", line 150, in mono_build_solution
    msbuild_info = find_msbuild_windows(env)
  File "D:\godot-master\build\godot\modules\mono\build_scripts\godotsharptools_build.py", line 106, in find_msbuild_windows
    mono_root = env['mono_prefix'] or find_mono_root_dir(env['bits'])
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Environment.py", line 410, in __getitem__
    return self._dict[key]
KeyError: 'mono_prefix'
scons: building terminated because of errors.
```

**Using**
`scons p=windows tools=yes module_mono_enabled=yes mono_glue=no -j3`

**Comment by exts** [comment]
In the docs @neikeq provided 

```
By default, SCons will try to find Mono in the Windows Registry on Windows or via ``pkg-config`` on other platforms.
You can specify a different installation directory by passing the ``mono_prefix`` command-line option to SCons;
e.g.: ``scons [...] mono_prefix=%ProgramFiles%/Mono``.


The specified directory must contain the subdirectories ``bin``, ``include``, and ``lib``.
This is the directory that contains the subdirectories ``include`` and ``lib``.
```

Does that mean I need to manually set mono_prefix now? https://github.com/godotengine/godot-docs/pull/2493

**Comment by akien-mga** [comment]
You should set `mono_prefix` from now on indeed. The config script should be changed to check that the option was set before accessing it though, and return an error if missing.

**Comment by exts** [comment]
Going to let this continue building, but for now I want to comment that `mono_prefix` does not play well with whitespace even when using `%ProgramFiles%` have to wrap the whole thing in a quotes or it'll think you're telling it to look inside `C:/Program` which doesn't exist.

Hopefully this solves my issue, current working command:

`scons p=windows tools=yes module_mono_enabled=yes mono_glue=no mono_prefix="C:/Program Files/Mono" -j3`

**Comment by exts** [comment]
Alright looks like the previous doesn't matter, as it got to that same part I get the same error:

```
[Initial build] mono_build_solution(["bin\GodotSharp\Tools\GodotSharpTools.dll"], ["modules\mono\editor\GodotSharpTools\GodotSharpTools.sln"])
[Initial build] scons: *** [bin\GodotSharp\Tools\GodotSharpTools.dll] KeyError : 'mono_prefix'
Traceback (most recent call last):
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Action.py", line 1197, in execute
sfnt.c
    result = self.execfunction(target=target, source=rsources, env=env)
  File "D:\godot-master\build\godot\modules\mono\build_scripts\godotsharptools_build.py", line 150, in mono_build_solution
    msbuild_info = find_msbuild_windows(env)
  File "D:\godot-master\build\godot\modules\mono\build_scripts\godotsharptools_build.py", line 106, in find_msbuild_windows
    mono_root = env['mono_prefix'] or find_mono_root_dir(env['bits'])
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-3.0.0\SCons\Environment.py", line 410, in __getitem__
    return self._dict[key]
KeyError: 'mono_prefix'
scons: building terminated because of errors.
```

**Comment by exts** [comment]
Did some deeper digging and it looks like mono_prefix isn't being updated in the Environment @ mono_configure.py I tested this by checking what was inside `env` before when the error called by `print env.Dump()`

@neikeq

**Comment by exts** [comment]
Couple things I messed with, not sure it's the correct solution, but I got it working.

- Downgraded mono back to 5.18.1. (5.20.x is missing nuget?)
- `set MONO_VERSION=5.18.1` in cmd
- edit `mono_configure.py` add `envvars.Update(env_mono)` under `envvars.Update(env)` (honestly not sure if it was just a typo, but i added it to both just in case. idk much about SCons.
- Ran: `scons p=windows tools=yes module_mono_enabled=yes mono_glue=no mono_prefix="C:/Program Files/Mono" -j3`

Build windows 10 editor.
